src/clj/game/cards/ice.clj

```diff
@@ -889,6 +889,35 @@
    "Galahad"
    (grail-ice end-the-run)
 
+   "Gatekeeper"
+   (let [draw {:prompt "Draw how many cards?"
```

Comment:
I think I was unclear; this `draw` ability definitely needs to be async so Jinja City Grid fully resolves before you move on to the shuffle phase of the subroutine. It's the `reveal-and-shuffle` ability that does not need to be, since moving cards and shuffling are both non-async functions.

Reply:
Ah okay! Thanks.